src/buffer/out/textBuffer.cpp

```diff
@@ -2085,187 +1956,326 @@ std::wstring TextBuffer::GetPlainText(const til::point& start, const til::point&
+                const auto fgHex = Utils::ColorToHexString(fg);
+                const auto bgHex = Utils::ColorToHexString(bg);
+                const auto ulHex = Utils::ColorToHexString(ul);
+                const auto ulStyle = attr.GetUnderlineStyle();
+                const auto isUnderlined = ulStyle != UnderlineStyle::NoUnderline;
+                const auto isCrossedOut = attr.IsCrossedOut();
+                const auto isOverlined = attr.IsOverlined();
+
+                htmlBuilder += "<SPAN STYLE=\"";
+                fmt::format_to(std::back_inserter(htmlBuilder), FMT_COMPILE("color:{};"), fgHex);
+                fmt::format_to(std::back_inserter(htmlBuilder), FMT_COMPILE("background-color:{};"), bgHex);
+
+                if (isIntenseBold && attr.IsIntense())
                 {
-                    // do not include \r nor \n as they don't have color attributes
-                    // and are not HTML friendly. For line break use '<BR>' instead.
-                    writeAccumulatedChars(false);
-                    break;
+                    htmlBuilder += "font-weight:bold;";
                 }
 
-                auto colorChanged = false;
-                if (!fgColor.has_value() || rows.FgAttr.at(row).at(col) != fgColor.value())
+                if (attr.IsItalic())
                 {
-                    fgColor = rows.FgAttr.at(row).at(col);
-                    colorChanged = true;
+                    htmlBuilder += "font-style:italic;";
                 }
 
-                if (!bkColor.has_value() || rows.BkAttr.at(row).at(col) != bkColor.value())
+                if (isCrossedOut || isOverlined)
                 {
-                    bkColor = rows.BkAttr.at(row).at(col);
-                    colorChanged = true;
+                    fmt::format_to(std::back_inserter(htmlBuilder),
+                                   FMT_COMPILE("text-decoration:{} {} {};"),
+                                   isCrossedOut ? "line-through" : "",
+                                   isOverlined ? "overline" : "",
+                                   fgHex);
                 }
 
-                if (colorChanged)
+                if (isUnderlined)
                 {
-                    writeAccumulatedChars(false);
+                    // Since underline, overline and strikethrough use the same css property,
+                    // we cannot apply different colors to them at the same time. However, we
+                    // can achieve the desired result by creating a nested <span> and applying
+                    // underline style and color to it.
+                    htmlBuilder += "\"><SPAN STYLE=\"";
 
-                    if (hasWrittenAnyText)
+                    switch (ulStyle)
                     {
-                        htmlBuilder << "</SPAN>";
+                    case UnderlineStyle::NoUnderline:
+                        break;
+                    case UnderlineStyle::SinglyUnderlined:

```

Comment:
nit: could move this down to 2204 and merge with `default:` like other underline consumers